src/core/thread/dua_manager.cpp

```diff
@@ -303,7 +303,18 @@ void DuaManager::UpdateCheckDelay(uint8_t aDelay)
 
 void DuaManager::HandleNotifierEvents(Events aEvents)
 {
-    Mle::MleRouter &mle = Get<Mle::MleRouter>();
+    Mle::MleRouter  &mle = Get<Mle::MleRouter>();
+    Lowpan::Context context;
+
+    if (aEvents.Contains(kEventThreadNetdataChanged))
+    {
+        // Remove a stale DUA address if any.
+        if (Get<ThreadNetif>().HasUnicastAddress(Get<DuaManager>().GetDomainUnicastAddress())
```

Comment:
May need to add `#if OPENTHREAD_CONFIG_DUA_ENABLE` around this?

Noticed builds are failing with this change. Looks like these method are only available under `OPENTHREAD_CONFIG_DUA_ENABLE. But at the top of `dua_manager.cpp` we have this
```c++
#if OPENTHREAD_CONFIG_DUA_ENABLE || (OPENTHREAD_FTD && OPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE)
```

Reply:
Thanks! Resolved your suggestions. Please verify.